src/llmcompressor/modifiers/pruning/sparsegpt/sgpt_base.py

```diff
@@ -191,21 +192,21 @@ def on_end(self, state: State, event: Event, **kwargs):
         self.ended_ = True
         self.remove_hooks()
 
-    def _infer_sequential_targets(
-        self, model: torch.nn.Module
-    ) -> Union[str, List[str]]:
+    def _infer_sequential_targets(self, model: torch.nn.Module) -> str | list[str]:
         if self.sequential_targets is None:
             return get_no_split_params(model)
-        if isinstance(self.sequential_targets, str):
-            return [self.sequential_targets]
-        return self.sequential_targets
+        match self.sequential_targets:
+            case str():
+                return [self.sequential_targets]
+            case _:
+                return self.sequential_targets
```

Comment:
Might be cleaner with the match if we include the prior if check as well
```suggestion
        match self.sequential_targets:
            case None:
                return get_no_split_params(model)
            case str():
                return [self.sequential_targets]
            case _:
                return self.sequential_targets
```